cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/LedgerEvents/Gov/DRepDeposits.hs

```diff
@@ -0,0 +1,226 @@
+
+  -- Start a local test net
+  conf@Conf { tempAbsPath } <- mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+      tempBaseAbsPath = makeTmpBaseAbsPath tempAbsPath
+
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  let sbe = ShelleyBasedEraConway
+      era = toCardanoEra sbe
+      cEra = AnyCardanoEra era
+      fastTestnetOptions = cardanoDefaultTestnetOptions
+        { cardanoEpochLength = 100
+        , cardanoSlotLength = 0.1
+        , cardanoNodeEra = cEra
+        , cardanoNumDReps = 0
+        }
+
+  TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    , wallets=wallet0:wallet1:_
+    , configurationFile
+    }
+    <- cardanoTestnetDefault fastTestnetOptions conf
+
+  poolNode1 <- H.headM poolNodes
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+  execConfig <- H.mkExecConfig tempBaseAbsPath poolSprocket1 testnetMagic
+
+  let socketName' = IO.sprocketName poolSprocket1
+      socketBase = IO.sprocketBase poolSprocket1 -- /tmp
+      socketPath = socketBase </> socketName'
+
+  epochStateView <- getEpochStateView (File configurationFile) (File socketPath)
+
+  H.note_ $ "Sprocket: " <> show poolSprocket1
+  H.note_ $ "Abs path: " <> tempAbsBasePath'
+  H.note_ $ "Socketpath: " <> socketPath
+  H.note_ $ "Foldblocks config file: " <> configurationFile
+
+  gov <- H.createDirectoryIfMissing $ work </> "governance"
+
+  -- DRep 1 (enough deposit)
+
+  drepDir1 <- H.createDirectoryIfMissing $ gov </> "drep1"
+
+  drepKeyPair1 <- generateDRepKeyPair execConfig drepDir1 "keys"
+  drepRegCert1 <- generateRegistrationCertificate execConfig drepDir1 "reg-cert"
+                                                  drepKeyPair1 1_000_000
+  drepRegTxBody1 <- createDRepRegistrationTxBody execConfig epochStateView sbe drepDir1 "reg-cert-txbody"
+                                                 drepRegCert1 wallet0
+  drepSignedRegTx1 <- signTx execConfig drepDir1 "signed-reg-tx"
+                             drepRegTxBody1 [drepKeyPair1, paymentKeyInfoPair wallet0]
+
+  submitTx execConfig drepSignedRegTx1
+
+  -- DRep 2 (not enough deposit)
+
+  drepDir2 <- H.createDirectoryIfMissing $ gov </> "drep2"
+
+  drepKeyPair2 <- generateDRepKeyPair execConfig drepDir2 "keys"
+  drepRegCert2 <- generateRegistrationCertificate execConfig drepDir2 "reg-cert"
+                                                  drepKeyPair2 999_999
+  drepRegTxBody2 <- createDRepRegistrationTxBody execConfig epochStateView sbe drepDir2 "reg-cert-txbody"
+                                                 drepRegCert2 wallet1
+  drepSignedRegTx2 <- signTx execConfig drepDir2 "signed-reg-tx"
+                             drepRegTxBody2 [drepKeyPair2, paymentKeyInfoPair wallet1]
+
+  failToSubmitTx execConfig drepSignedRegTx2
+
+
+
+-- DRep key pair generation
+
+generateDRepKeyPair :: (MonadTest m, MonadCatch m, MonadIO m, HasCallStack) => H.ExecConfig -> FilePath -> String -> m PaymentKeyPair
```

Comment:
We need a separate module to host all key generation functions for cardano-testnet.

Reply:
I will put it in `Testnet.Components.DReps` for the moment